rfd/0237-sub-ca-support.md

```diff
@@ -0,0 +1,677 @@
+---
+authors: Alan Parra (alan.parra@goteleport.com)
+state: draft
+---
+
+# RFD 0237 - Sub CA support
+
+## Required Approvers
+
+* Engineering: @rosstimothy && @espadolini && @greedy52 && @nklaassen
+* Security: (@rob-picard-teleport || @klizhentas)
+* Product: @klizhentas
+
+## What
+
+Add support for Teleport CAs to operate as a sub CA of an external root.
+
+While the feature design is generic, support is (initially) only available for a
+subset of Teleport CAs that are commonly visible to external trusted root
+configurations: namely the "DB client" and "Desktop Access" CAs.
+
+Sub CA support is an Enterprise feature.
+
+## Why
+
+Teleport operates multiple distinct CAs, each used to mint certificates for its
+supported protocols. Historically those CAs all operate as self-signed roots, a
+simple and effective solution.
+
+As Teleport's customer base grows it becomes more common to encounter
+corporations that want a larger degree of control over those CAs. The sub CA
+feature addresses internal security policies and allows organizations to chain
+the more "visible" Teleport CAs to their own self-managed roots.
+
+## UX
+
+<a id="ux1"></a>
+### Alice configures "db-client" as a sub CA
+
+1. First, Alice issues CSRs for the desired CA / cluster.
+
+    ```shell
+    $ tctl auth sub-ca create-csr --type=db-client
```

Comment:
Playing devils advocate a bit here, does the sub-ca subcommand add value? We don't have `tctl auth ca rotate` to initiate a CA rotation, so why do sub CAs require a special subcommand? Do the new commands proposed in this RFD still make sense if we remove the sub-ca command?


```bash

tctl auth create-csr --type=db-client

tctl auth create-override --type=db-client cert.pem [chain.pem...]

tctl auth delete-override --type=db-client

```

Reply:
Maybe `tctl auth create-override-csr`, so the commands are neatly grouped?